telegramBot.js

```diff
@@ -0,0 +1,200 @@
+
+            ctx.reply(helpText)
+        })
+
+        this.bot.command('quit', (ctx) => {
+            ctx.leaveChat()
+        })
+        
+        this.bot.command('gm', async (ctx) => {
+            if(this.initialzied) {
+
+                if(this.onCommandReceived){
+                    const input = ctx.message.text.slice(1).trim().split(' ');
+                    const command = input.shift();
+                    const commandArgs = input.join(' ');
+                    const result = await this.onCommandReceived(command, commandArgs, 'telegram');
+                    console.log(result);
+                    return result;
+                }
+                
+            } else {
+                ctx.reply('Bot uninitialized. Run /init command to initialize.')
+            }
+        })
+        
+        this.bot.command('fs', async (ctx) => {
+            if(this.initialzied)  {
+                if(this.onCommandReceived){
+                    const input = ctx.message.text.slice(1).trim().split(' ');
+                    const command = input.shift();
+                    const commandArgs = input.join(' ');
+                    const result = await this.onCommandReceived(command, commandArgs, 'telegram');
+                    console.log(result);
+                    return result;
+                }
+            
+                const fileURL = "http://i.imgur.com/5rOhtdL.png"
+                this.sendFile(fileURL)
+            } else {
+                ctx.reply('Bot uninitialized. Run /init command to initialize.')
+            }
+        })
+
+        this.bot.command('fs_r', async (ctx) =>{
+            if(this.initialzied)  {
+                if(this.onCommandReceived){
+                    const input = ctx.message.text.slice(1).trim().split(' ');
+                    const command = input.shift();
+                    const commandArgs = input.join(' ');
+                    const result = await this.onCommandReceived(command, commandArgs, 'telegram');
+                    console.log(result);
+                    return result;
+                }
+            
+                const fileURL = 'https://homepages.cae.wisc.edu/~ece533/images/airplane.png'
+                this.sendFileRemote(fileURL)
+            } else {
+                ctx.reply('Bot uninitialized. Run /init command to initialize.')
+            }
+        })
+
+        this.bot.use((ctx) => {
+            this.chatID = ctx.chat.id
+            this.chatTitle = ctx.chat.title
+        })
+
+    }
+
+    setCommandListener(commandListener) {
+        this.onCommandReceived = commandListener;
+    }
+
+    async getUpdates(){
+
+        try{
+            const update = await axios.get(this.updateURL);
+            return update.data;  
+        } catch(err){
+            console.log(err)
+        }
+        
+    }
+
+    getChannels(){
+        return {
```

Comment:
this function should return all of the groups the bot has, instead of just return the `chatID` since this is only for the current chat. But looks like the telegram doesn't have api to get all group, we need to find some workaround in future

Reply:
https://github.com/rubenlagus/TelegramBots/issues/297